mkosi

```diff
@@ -2492,6 +2492,20 @@ def set_root_password(args: CommandLineArguments, root: str, do_run_build_script
             patch_file(os.path.join(root, 'etc/shadow'), jj)
 
 
+def set_serial_terminal(args: CommandLineArguments, root: str, do_run_build_script: bool, for_cache: bool) -> None:
+    """Override TERM for the serial console with the terminal type from the host."""
+
+    if do_run_build_script or for_cache or not args.qemu_headless:
+        return
+
+    override_dir = mkdir_last(os.path.join(root, "etc/systemd/system/serial-getty@ttyS0.service.d"))
+    with open(os.path.join(override_dir, "override.conf"), 'w') as f:
+        f.write(f"""
+[Service]
+Environment=TERM={os.getenv('TERM')}
```

Comment:
This might be a silly question, but is TERM always guaranteed to be set (e.g. in CI systems)? If not a default for getenv would be a good idea.

Reply:
Good point, we now default to vt220 which is what systemd does as well.